src/index.ts

```diff
@@ -856,7 +854,7 @@ class Spanner extends GrpcService {
       const requestFn = gaxClient[config.method].bind(
         gaxClient,
         reqOpts,
-        config.gaxOpts
+        addResourcePrefixHeader(config.gaxOpts, config.headers)
```

Comment:
I think we should just do the extend within this function, rather than have the very specific, single-use external method. It will be more clear for someone to see how it works.